example/generator/odp_generator.c

```diff
@@ -1169,12 +1169,10 @@ int main(int argc, char *argv[])
 	/* Print both system and application information */
 	print_info(NO_PATH(argv[0]), &args->appl);
 
-	/* Default to max number of workers, unless user specified number of
-	 * workers or cpumask */
 	num_workers = MAX_WORKERS;
```

Comment:
Set default num_workers to 1.... no need to complicate '-w' or '-c' processing for this.

-       /* Default to max number of workers, unless user specified number of
+       /* Default to 1 worker, unless user specified number of
         * workers or cpumask */
-       num_workers = MAX_WORKERS;
+       num_workers = 1;

Reply:
The added complexity is required to handle case `-w 0` -> use all available cores. Setting num_workers to MAX_WORKERS inside parse_args() could perhaps be cleaner.

Reply:
In my view, I see the setting of number of workers like this:
- one default: MAX_WORKERS (now) or 1 in your case
- two overrides:
         '-w':  override default by specify the number of workers as number
         '-c':  override default by specify the number of workers/thread distribution as core mask
- '-w' has precedence over '-c'

As you said, -w 0 can be converted to 'appl_args->num_workers = MAX_WORKERS'.

Yet, I still believe we should run with maximum performance (worker count) instead of minimum.